### PR TITLE
Fixes  #8154 - Add collect endpoint for web analytic events

### LINKED ISSUE #8154: Create collect endpont for data insight

### PR DESCRIPTION
**Pull request by TeddyCr**
### Describe your changes :
Fixes #8154 by adding a collect endpoint to send web analytic events.

Note -- this PR branched off https://github.com/open-metadata/OpenMetadata/pull/8122. #8122 will need to be merged prior to merging this PR. Below are the commits that are part of this PR specifically and will need to be reviewed:
- https://github.com/open-metadata/OpenMetadata/pull/8193/commits/abbd208aa37ee79910d6e4bce1846358e9236bc2
- https://github.com/open-metadata/OpenMetadata/pull/8193/commits/b3436bb76766dfa2395dcf3b00a96a1b915ac4a8
- https://github.com/open-metadata/OpenMetadata/pull/8193/commits/51bfaa229312e4cc15bd8fa7d2b0e2f1dba3a6d8

#
### Type of change :

- [x] Bug fix
- [x] Improvement
- [x] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers
Backend: @open-metadata/backend

### CODE REVIEW

File: bootstrap/sql/com.mysql.cj.jdbc.Driver/v006__create_db_connection_info.sql [hunk 1]
```diff
@@ -0,0 +1,23 @@
+CREATE TABLE IF NOT EXISTS report_definition (
+    id VARCHAR(36) GENERATED ALWAYS AS (json ->> '$.id') NOT NULL,
+    name VARCHAR(256) GENERATED ALWAYS AS (json ->> '$.name') NOT NULL,
+    fullyQualifiedName VARCHAR(256) GENERATED ALWAYS AS (json ->> '$.fullyQualifiedName') NOT NULL,
+    json JSON NOT NULL,
+    updatedAt BIGINT UNSIGNED GENERATED ALWAYS AS (json ->> '$.updatedAt') NOT NULL,
+    updatedBy VARCHAR(256) GENERATED ALWAYS AS (json ->> '$.updatedBy') NOT NULL,
+    deleted BOOLEAN GENERATED ALWAYS AS (json -> '$.deleted'),
+    UNIQUE (name)
+);
```
Comment (harshach) [inline bootstrap/sql/com.mysql.cj.jdbc.Driver/v006__create_db_connection_info.sql:10]: add indexes around these tables

**Comment by Sachin-chaurasiya** [comment]
> @Sachin-chaurasiya for reference


Thanks @TeddyCr

File: bootstrap/sql/com.mysql.cj.jdbc.Driver/v006__create_db_connection_info.sql [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    name VARCHAR(256) GENERATED ALWAYS AS (json ->> '$.name') NOT NULL,
+    fullyQualifiedName VARCHAR(256) GENERATED ALWAYS AS (json ->> '$.fullyQualifiedName') NOT NULL,
+    eventType VARCHAR(256) GENERATED ALWAYS AS (json ->> '$.eventType') NOT NULL,
+    json JSON NOT NULL,
+    updatedAt BIGINT UNSIGNED GENERATED ALWAYS AS (json ->> '$.updatedAt') NOT NULL,
+    updatedBy VARCHAR(256) GENERATED ALWAYS AS (json ->> '$.updatedBy') NOT NULL,
+    deleted BOOLEAN GENERATED ALWAYS AS (json -> '$.deleted'),
+    UNIQUE(name),
+    INDEX name_index (name),
+    INDEX event_type_index (eventType)
```
Comment (harshach) [inline bootstrap/sql/com.mysql.cj.jdbc.Driver/v006__create_db_connection_info.sql:25]: I think we can remove the eventType. If the column is low cardinality that is there are only few unique values indexing doesn't work well in MySQL and ends up causing perf issues

File: openmetadata-service/src/main/java/org/openmetadata/service/resources/analytics/AnalyticsReportResource.java [hunk 3]
```diff
@@ -0,0 +1,446 @@
...
+  public void initialize(OpenMetadataApplicationConfig config) throws IOException {
+    // Find the existing reportDefinition and them from json files
+    List<String> reportDefinitionFiles =
+        EntityUtil.getJsonDataResources(".*json/data/analytics/reportDefinitions/.*\\.json$");
+    reportDefinitionFiles.forEach(
+        reportDefinitionFile -> {
+          try {
+            String reportDefinitionJson =
+                CommonUtil.getResourceAsStream(getClass().getClassLoader(), reportDefinitionFile);
+            reportDefinitionJson = reportDefinitionJson.replace("<separator>", Entity.SEPARATOR);
```
Comment (harshach) [inline openmetadata-service/src/main/java/org/openmetadata/service/resources/analytics/AnalyticsReportResource.java:83]: final static string for <separator>

File: openmetadata-service/src/main/java/org/openmetadata/service/resources/analytics/WebAnalyticEventResource.java [hunk 4]
```diff
@@ -0,0 +1,447 @@
...
+import io.swagger.v3.oas.annotations.parameters.RequestBody;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import java.io.IOException;
+import java.util.List;
+import java.util.UUID;
+import javax.json.JsonPatch;
+import javax.validation.Valid;
+import javax.validation.constraints.Max;
+import javax.validation.constraints.Min;
+import javax.ws.rs.*;
```
Comment (harshach) [inline openmetadata-service/src/main/java/org/openmetadata/service/resources/analytics/WebAnalyticEventResource.java:22]: avoid wild-cards imports

File: openmetadata-service/src/main/java/org/openmetadata/service/resources/analytics/WebAnalyticEventResource.java [hunk 5]
```diff
@@ -0,0 +1,447 @@
...
+    public WebAnalyticEventDataList(
+        List<WebAnalyticEventData> data, String beforeCursor, String afterCursor, int total) {
+      super(data, beforeCursor, afterCursor, total);
+    }
+  }
+
+  @SuppressWarnings("unused") // Method used for reflection of webAnalyticEventTypes
+  public void initialize(OpenMetadataApplicationConfig config) throws IOException {
+    // Find the existing reportDefinition and them from json files
+    List<String> webAnalyticEventFiles =
```
Comment (harshach) [inline openmetadata-service/src/main/java/org/openmetadata/service/resources/analytics/WebAnalyticEventResource.java:95]: same as before

**Pull request merged**